test/parallel/test-child-process-send-type-error.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+require('../common');
+const assert = require('assert');
+const cp = require('child_process');
+
+function noop() {}
+
+function fail(proc, args) {
+  assert.throws(() => {
+    proc.send.apply(proc, args);
+  }, /"options" argument must be an object/);
+}
+
+if (process.argv[2] !== 'child') {
+  const child = cp.fork(__filename, ['child']);
+
+  fail(child, ['msg', null, null]);
+  fail(child, ['msg', null, '']);
+  fail(child, ['msg', null, 'foo']);
+  fail(child, ['msg', null, 0]);
+  fail(child, ['msg', null, NaN]);
+  fail(child, ['msg', null, 1]);
+  fail(child, ['msg', null, null, noop]);
+} else {
+  fail(process, ['msg', null, null]);
+  fail(process, ['msg', null, '']);
+  fail(process, ['msg', null, 'foo']);
+  fail(process, ['msg', null, 0]);
+  fail(process, ['msg', null, NaN]);
+  fail(process, ['msg', null, 1]);
+  fail(process, ['msg', null, null, noop]);
+}
```

Comment:
This could be made a little less redundant if you write it like this:
```javascript
let target = process;
if (process.argv[2] !== 'child')
  target = cp.fork(__filename, ['child']);
fail(target, ['msg', null, null]);
// ...
```
Just a suggestion, though.